code-submissions-clustering-core/src/main/kotlin/org/jetbrains/research/code/submissions/clustering/model/SubmissionsGraph.kt

```diff
@@ -0,0 +1,36 @@
+package org.jetbrains.research.code.submissions.clustering.model
+
+import org.jetbrains.research.code.submissions.clustering.load.AbstractUnifier
+import org.jgrapht.Graph
+import org.jgrapht.graph.DefaultWeightedEdge
+import org.jgrapht.graph.SimpleDirectedWeightedGraph
+
+data class SubmissionsGraph(private val graph: Graph<SubmissionsNode, DefaultWeightedEdge>) {
+    fun buildStringRepresentation() = graph.toString()
+}
+
+class GraphBuilder(private val unifier: AbstractUnifier) {
+    private val graph: Graph<SubmissionsNode, DefaultWeightedEdge> =
+        SimpleDirectedWeightedGraph(DefaultWeightedEdge::class.java)
+
+    fun add(submission: Submission) {
+        unifier.run {
+            val unifiedSubmission = submission.unify()
+            // Check if suitable vertex already exists
+            val vertex = graph.vertexSet().find { it.code == unifiedSubmission.code }
```

Comment:
For big graphs it will works slow. Maybe is it better to make a cache with hashes of all vertices? Or just add todo about it, we definitely should add it in the future

Reply:
Added a hashmap for storing added vertices